client/src/components/ContentNavigator/viya/ContentModel.ts

```diff
@@ -0,0 +1,417 @@
+
+    return result.items.map((childItem: ContentItem) => ({
+      ...childItem,
+      uid: `${childItem.id}${item.name}`.replace(/\s/g, ""),
+      __trash__: isTrash,
+    }));
+  }
+
+  public async getParent(item: ContentItem): Promise<ContentItem | undefined> {
+    const ancestorsLink = getLink(item.links, "GET", "ancestors");
+    if (!ancestorsLink) {
+      return;
+    }
+
+    const resp = await this.connection.get(ancestorsLink.uri);
+    if (resp.data.ancestors && resp.data.ancestors.length > 0) {
+      return resp.data.ancestors[0];
+    }
+  }
+
+  public async getResourceByUri(uri: Uri): Promise<ContentItem> {
+    const resourceId = getResourceId(uri);
+    const res = await this.connection.get(resourceId);
+    this.fileTokenMaps[resourceId] = {
+      etag: res.headers.etag,
+      lastModified: res.headers["last-modified"],
+    };
+
+    return res.data;
+  }
+
+  public async getContentByUri(uri: Uri): Promise<string> {
+    const resourceId = getResourceId(uri);
+    const res = await this.connection.get(resourceId + "/content", {
+      transformResponse: (response) => response,
+    });
+
+    this.fileTokenMaps[resourceId] = {
+      etag: res.headers.etag,
+      lastModified: res.headers["last-modified"],
+    };
+
+    // We expect the returned data to be a string. If this isn't a string,
+    // we can't really open it
+    if (typeof res.data === "object") {
+      throw new Error(Messages.FileOpenError);
+    }
+
+    return res.data;
+  }
+
+  public async createFile(
+    item: ContentItem,
+    fileName: string
+  ): Promise<ContentItem | undefined> {
+    const contentType = await this.getFileContentType(fileName);
+
+    let fileCreationResponse = null;
+    try {
+      fileCreationResponse = await this.connection.post(
+        `/files/files#rawUpload?typeDefName=${contentType}`,
+        Buffer.from("", "binary"),
+        {
+          headers: {
+            "Content-Type": "text/plain",
+            "Content-Disposition": `filename="${fileName}"`,
+          },
+        }
+      );
+    } catch (error) {
+      return;
+    }
+
+    const fileLink: Link | null = getLink(
+      (fileCreationResponse.data as ContentItem).links,
+      "GET",
+      "self"
+    );
+
+    const memberAdded = await this.addMember(
+      fileLink?.uri,
+      getLink(item.links, "POST", "addMember")?.uri,
+      {
+        name: fileName,
+        contentType,
+      }
+    );
+    if (!memberAdded) {
+      return;
+    }
+
+    return fileCreationResponse.data;
+  }
+
+  public async createFolder(
+    item: ContentItem,
+    name: string
+  ): Promise<ContentItem | undefined> {
+    const parentFolderUri =
+      item.uri || getLink(item.links || [], "GET", "self")?.uri || null;
+    if (!parentFolderUri) {
+      return;
+    }
+
+    try {
+      const createFolderResponse = await this.connection.post(
+        `/folders/folders?parentFolderUri=${parentFolderUri}`,
+        {
+          name,
```

Comment:
"My Favorites" folder and other user created favorite folders only allow to create favorite folders.
{
    name,
    type:  this.dataDescriptor.getTypeName(parentItem) === "favoritesFolder" ?
                    "favoritesFolder" : "folder"
}